library/src/main/java/com/novoda/downloadmanager/lib/DownloadNotifier.java

```diff
@@ -284,13 +288,15 @@ private Notification buildTitlesAndDescription(
             long totalBytes = 0;
             long totalBytesPerSecond = 0;
             synchronized (mDownloadSpeed) {
-                for (DownloadInfo info : cluster) {
-                    if (info.mTotalBytes != -1) {
-                        currentBytes += info.mCurrentBytes;
-                        totalBytes += info.mTotalBytes;
-                        Long bytesPerSecond = mDownloadSpeed.get(info.batchId);
-                        if (bytesPerSecond != null) {
-                            totalBytesPerSecond += bytesPerSecond;
+                for (ActiveBatch batch : cluster) {
+                    for (DownloadInfo info : batch.getDownloads()) {
+                        if (info.mTotalBytes != -1) {
```

Comment:
or maybe better a method on the `DownloadInfo`, `info.hasTotalBytes()` would it work better?